src/js/strategies/CitrixVDIStrategy.js

```diff
@@ -6,28 +6,41 @@ import CCPInitiationStrategyInterface from "./CCPInitiationStrategyInterface";
 import {hitch} from "../utils";
 import {FailedState} from "../rtc_session";
 import {RTC_ERRORS} from "../rtc_const";
-import "@citrix/ucsdk/CitrixWebRTC";
+
+
 
 export default class CitrixVDIStrategy extends CCPInitiationStrategyInterface {
 
-    constructor() {
+    constructor(useRealCitrix = true) {
         super();
-        if (!window.CitrixWebRTC.isFeatureOn("webrtc1.0")) {
-            throw new Error('Citrix WebRTC redirection feature is NOT supported!');
-        }
-        window.getCitrixWebrtcRedir = function () {
-            const registryValue = Promise.resolve(1);
-            return new Promise(function (resolve, reject) {
-                //retrieve registry value internally
-                registryValue.then((v) => {
-                    resolve(v);
-                }).catch(() => {
-                    reject();
-                })
-            })
+        if(useRealCitrix){
+            require("@citrix/ucsdk/CitrixWebRTC");
         }
+        console.log("CitrixVDIStrategy initializing");
+        this.initCitrixWebRTC();
+        this.initGetCitrixWebrtcRedir();
+        this.initLog();
+    }
+
+    initCitrixWebRTC() {
+        window.CitrixWebRTC.setVMEventCallback((event) => {
+            if (event.event === 'vdiClientConnected') {
```

Comment:
What other events do we get from Citrix? Would it logging other events be helpful in troubleshooting?

Reply:
when setup callback before initUCSDK, it is intended to be used for checking initialization success or failure status.
The rest of the fields are relevant to OS info that we have via userAgent in call end report, and citrix versino info. 

```
{ "event":"vdiClientConnected",
"type_script": { "type":5, "version":"1.6.0.1"},
"webrpc": {"type":0, "version":"1.6.0."},
"webrtc_codecs":{"type":1, "version":"1.76.0.0"},
"receiver": {"type":2, "version":"19.12.0.14"},
"vda": {"type":3, "version":"0.0.0.0"},
"endpoint": {"type":4, "version":"0.0.0.0"}
"osinfo": { "architecture":"x86_64",
"distro":"Microsoft",
"edition":"Windows 10 Enterprise\u0000",
"family":"Windows",
"version":"6.3\u0000"}
}
```

Reply:
I think most of those version info is also printed in the Citrix log, and we collected it in CCP Log